clang/unittests/Format/FormatTest.cpp

```diff
@@ -22537,10 +22537,12 @@ TEST_F(FormatTest, FormatsLambdas) {
                "  }\n"
                "}",
                Style);
-  verifyFormat("std::sort(v.begin(), v.end(),\n"
-               "          [](const auto &foo, const auto &bar) {\n"
-               "  return foo.baz < bar.baz;\n"
-               "});",
```

Comment:
Can you put it back in? In general, we want to keep existing test cases.

Reply:
I can add this back in but the reason I removed it in favour of putting the equivalent code inside block scope is because the code in this example is a violation of C++ grammar given it is not actually a declaration but is at namespace/global scope. As such, the new logic incorrectly assumes that this is a declaration and would therefore would **not** indent the lambda body wrt. the start of the expression, which I think would be misleading to readers.

Simply putting this code inside block scope captures the intention of original test case, I think.

Reply:
> I can add this back in but the reason I removed it in favour of putting the equivalent code inside block scope is because the code in this example is a violation of C++ grammar given it is not actually a declaration but is at namespace/global scope.

Yeah, but many clang-format test cases (`return` statements, function calls, etc) seem "invalid" because they are not wrapped in a block. What if the test case above is in a macro definition?

Reply:
I've added a test which puts the `std::sort(...)` in a macro definition. It works as expected.

I haven't added the original test back since I still think it's misleading but if you still think I should keep it I will add it back (and adjust it to pass as-is). Just let me know.